pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ExtractTransformFunction.java

```diff
@@ -0,0 +1,97 @@
+package org.apache.pinot.core.operator.transform.function;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+import org.joda.time.Chronology;
+import org.joda.time.DateTimeField;
+import org.joda.time.chrono.ISOChronology;
+
+
+public class ExtractTransformFunction extends BaseTransformFunction {
+  public static final String FUNCTION_NAME = "extract";
+  private TransformFunction _mainTransformFunction;
+  protected String _field;
+  protected Chronology _chronology = ISOChronology.getInstanceUTC();
+  
+  @Override
+  public String getName() { return FUNCTION_NAME; }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+    if (arguments.size() != 2) { 
+      throw new IllegalArgumentException("Exactly 2 arguments are required for EXTRACT transform function"); 
+    }
+
+    _field = ((LiteralTransformFunction) arguments.get(0)).getLiteral();
+    if(!validField(_field)) {
```

Comment:
(code format) Please reformat using the [Pinot Style](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#intellij)

Reply:
Thanks! I wanted to bring to your notice that original code is also getting reformatted while using the Pinot Style.